### PR TITLE
Add release notes for 2.7.0

### PR DESCRIPTION
**Pull request by madkinsz**
[Rendered](https://github.com/PrefectHQ/prefect/blob/release-notes-2.7.0/RELEASE-NOTES.md#release-270)

### CODE REVIEW

**Review by tpdorsey — COMMENTED** [review]
Looks good

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,5 +1,98 @@
...
+Note this process is robust to agent restarts, but does require an agent to be running for cancellation to be enforced.
+
+Support for cancellation has been added to all core library infrastructure types:
+- Docker Containers (https://github.com/PrefectHQ/prefect/pull/7684)
+- Kubernetes Jobs (https://github.com/PrefectHQ/prefect/pull/7701)
+- Processes (https://github.com/PrefectHQ/prefect/pull/7635)
+
+Cancellation support is in progress for all collection infrastructure types:
+- ECS Tasks (https://github.com/PrefectHQ/prefect-aws/pull/163)
+- Azure Container Instances (...)
```
Comment (rpeden) [inline RELEASE-NOTES.md:24]:
```suggestion
- Azure Container Instances (https://github.com/PrefectHQ/prefect-azure/pull/58)
```

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,5 +1,92 @@
...
+- Google Cloud Run Jobs (...)
+
+At this time, this feature requires the flow run to be submitted by an agent — flow runs without deployments cannot be cancelled yet. We're working on adding another cancellation mechanism that does not require the agent, but it will not provide as strong of a guarantee that your infrastructure shutsdown.
```
[the remark this answers is not in the document]
Comment (madkinsz) [reply RELEASE-NOTES.md:27]: I can add a link to the WIP for people to follow? I'm also okay removing it. cc @billpalombi

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,5 +1,92 @@
...
+...
+
+See https://github.com/PrefectHQ/prefect/pull/7361 for more details
+
+### Enhancements
+- Update login to prompt for "API key" instead of "authentication key" — https://github.com/PrefectHQ/prefect/pull/7649
+- Add `TwilioSMS` notification block — https://github.com/PrefectHQ/prefect/pull/7685
+- Add jitter to the agent query loop — https://github.com/PrefectHQ/prefect/pull/7652
+- Include final state logs in logs sent to API — https://github.com/PrefectHQ/prefect/pull/7647
+- Add `tags` to and `idempotency_key` to `run deployment` — https://github.com/PrefectHQ/prefect/pull/7641
```
[the remark this answers is not in the document]
Comment (madkinsz) [reply RELEASE-NOTES.md:63]:
```suggestion
- Add `tags` and `idempotency_key` to `run deployment` — https://github.com/PrefectHQ/prefect/pull/7641
```

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -1,5 +1,92 @@
...
+- Add GitLab storage block docs to deployment CLI — https://github.com/PrefectHQ/prefect/pull/7686
+- Add links to Cloud Run and Container Instance infrastructure — https://github.com/PrefectHQ/prefect/pull/7690
+- Update docs on final state determination to reflect `Cancelled` state changes — https://github.com/PrefectHQ/prefect/pull/7700
+- Fix link in 'Agents and Work Queues' documentation — https://github.com/PrefectHQ/prefect/pull/7659
+
+### Contributors
+- @brian-pond made their first contribution in https://github.com/PrefectHQ/prefect/pull/7659
+- @YtKC made their first contribution in https://github.com/PrefectHQ/prefect/pull/7641
+- @eudyptula made their first contribution in https://github.com/PrefectHQ/prefect/pull/7361
+- @brian-pond
```
[the remark this answers is not in the document]
Comment (madkinsz) [reply RELEASE-NOTES.md:91]: Nope! Will fix thanks!

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -1,5 +1,154 @@
...
+- Kubernetes Jobs (https://github.com/PrefectHQ/prefect/pull/7701)
+- Processes (https://github.com/PrefectHQ/prefect/pull/7635)
+
+Cancellation support is in progress for all collection infrastructure types:
+
+- ECS Tasks (https://github.com/PrefectHQ/prefect-aws/pull/163)
+- Google Cloud Run Jobs (https://github.com/PrefectHQ/prefect-gcp/pull/76)
+- Azure Container Instances (https://github.com/PrefectHQ/prefect-azure/pull/58)
+
+At this time, this feature requires the flow run to be submitted by an agent — flow runs without deployments cannot be cancelled yet but that feature is [coming soon](https://github.com/PrefectHQ/prefect/pull/7150).
```
Comment (tpdorsey) [inline RELEASE-NOTES.md:35]:
```suggestion
At this time, this feature requires the flow run to be submitted by an agent — flow runs without deployments cannot be cancelled yet, but that feature is [coming soon](https://github.com/PrefectHQ/prefect/pull/7150).
```

**Review by tpdorsey — APPROVED** [review]

**Pull request merged**